### PR TITLE
[KernelGen][Nvidia] Add persistent_topk operator with Triton kernel for VLLM

### PR DESCRIPTION
**Pull request by yzw1128**
## Summary

Add a Triton implementation of `persistent_topk` for DeepSeek-V4 sparse attention inference. This operator selects the top-k indices from each row of a logits tensor using a workspace-cached ternary search algorithm (4-way narrowing, 14 passes = 28 effective bisections). It replaces the vLLM CUDA `persistent_topk` kernel with a portable Triton implementation.

## Algorithm

1. **Phase 1**: Cache row data to workspace for L2 locality, compute min/max.
2. **Phase 2**: Ternary search (14 iterations) to find threshold value.
3. **Phase 3**: Scatter indices where value > threshold.
4. **Phase 4**: Fill remaining slots with elements equal to threshold.

Falls back to direct HBM reads when workspace is too small for caching.

## Correctness Verification

All test shapes pass correctness check against vLLM CUDA baseline (`torch.ops._C.persistent_topk`).
Comparison method: sort selected values descending and assert_close (atol=1e-6, rtol=1e-6).

| num_rows | seq_len | k | Result |
|----------|---------|------|--------|
| 1 | 1024 | 1024 | PASS |
| 1 | 2048 | 1024 | PASS |
| 1 | 4096 | 1024 | PASS |
| 4 | 4096 | 1024 | PASS |
| 8 | 8192 | 1024 | PASS |
| 16 | 16384 | 1024 | PASS |
| 32 | 32768 | 1024 | PASS |
| 4 | 4096 | 512 | PASS |
| 4 | 4096 | 2048 | PASS |

## Performance vs VLLM (NVIDIA H20)

| Configuration | Triton (us) | vLLM CUDA (us) | Speedup |
|---------------|-------------|----------------|---------|
| decode-1-1k (1, 1024, k=1024) | 28.5 | 18.1 | 0.64x |
| decode-1-2k (1, 2048, k=1024) | 28.2 | 23.2 | 0.82x |
| decode-1-4k (1, 4096, k=1024) | 28.9 | 17.5 | 0.61x |
| decode-4-4k (4, 4096, k=1024) | 29.4 | 19.9 | 0.68x |
| decode-8-8k (8, 8192, k=1024) | 56.2 | 23.8 | 0.42x |
| batch-16-16k (16, 16384, k=1024) | 107.8 | 21.6 | 0.20x |
| batch-32-32k (32, 32768, k=1024) | 217.8 | 24.2 | 0.11x |
| decode-4-4k-k512 (4, 4096, k=512) | 29.3 | 20.1 | 0.69x |
| decode-4-4k-k2048 (4, 4096, k=2048) | 32.3 | 19.1 | 0.59x |

Geometric mean speedup: 0.53x

## Files Changed

- **New**: `src/flag_gems/fused/persistent_topk.py` — Triton kernel implementation
- **New**: `tests/test_persistent_topk.py` — Correctness tests vs vLLM CUDA
- **New**: `benchmark/test_persistent_topk.py` — Performance benchmark
- **Modified**: `src/flag_gems/fused/__init__.py` — Register operator import
- **Modified**: `conf/operators.yaml` — Add operator metadata

### CODE REVIEW

File: tests/test_persistent_topk.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        (1, 4096, 1024),
+        (4, 4096, 1024),
+        (8, 8192, 1024),
+        (16, 16384, 1024),
+        (32, 32768, 1024),
+        (4, 4096, 512),
+        (4, 4096, 2048),
+    ]
+
+
```
Comment (tengqm) [inline tests/test_persistent_topk.py:63]:
```
@pytest.mark.persistent_topk
```

File: conf/operators.yaml [hunk 2]
```diff
@@ -7047,6 +7047,18 @@ ops:
       - NeuralNetwork
     stages:
       - beta: '5.3'
+  - id: persistent_topk
```
Comment (tengqm) [inline conf/operators.yaml:7050]: Please help keep the list sorted, `persistent_topk` is not supposed to appear after `top_k_per_row_decode`.

**Comment by huangyiqun** [comment]
Are you still working on this PR?
